detox/src/devices/android/ADB.js

```diff
@@ -96,13 +126,86 @@ class ADB {
   async adbCmd(deviceId, params) {
     const serial = `${deviceId ? `-s ${deviceId}` : ''}`;
     const cmd = `${this.adbBin} ${serial} ${params}`;
-    return await exec(cmd, undefined, undefined, 1);
+    return await execWithRetriesAndLogs(cmd, undefined, undefined, 1);
   }
 
   async sleep(ms = 0) {
     return new Promise((resolve, reject) => setTimeout(resolve, ms));
   }
 
+  async getScreenSize(deviceId) {
+    const {stdout} = await this.adbCmd(deviceId, `shell wm size`);
+    const [width, height] = stdout.split(' ').pop().split('x');
+    return {
+      width: parseInt(width, 10),
+      height: parseInt(height, 10)
+    };
+  }
+
+  async getFileSize(deviceId, path) {
+    const {stdout} = await this.adbCmd(deviceId, `shell wc -c ${path}`);
+    return parseInt(stdout, 10);
+  }
+
+  screencap(deviceId, path) {
```

Comment:
is this supposed to be an async function ?

Reply:
Yep. Missed. Thanks.